lib/TestScriptRunnable.rb

```diff
@@ -56,9 +56,29 @@ def load_fixtures
 
       hash[fixture.id] = resource
       type = resource.resourceType
+      
+      # Need to consider where to put this code, such as a separate method before setup execition.
+      if fixture.autocreate
+        FHIR.logger.info "[.load_fixture] Autocreate Fixture: #{fixture.id}"
+        script.setup.action.each do |ac|
+            next if ac.operation == nil
+            if ac.operation.sourceId == fixture.id
+              FHIR.logger.warn "Possible duplication of fixture creation (autocreation and setup section)"
+              FHIR.logger.warn "Check setup in TestScript. Operation type: #{ac.operation.type.code} , sourceId: #{ac.operation.sourceId}"
+            end
+        end
 
-      script.setup.action.unshift action_create(fixture.id, type) if fixture.autocreate
-      script.teardown.action << action_delete(fixture.id, type) if fixture.autodelete
+        begin
+          client.create(get_resource_from_ref(fixture.resource))
+        rescue StandardError => e
+          log_error e.message
+          report.error e.message
+          throw :exit
```

Comment:
```suggestion
```
The `:exit` throwing-catching a) is something I think we're gonna try to work out of the code anyways since it's kinda a code smell and b) does not apply to this case as this code is not called within the context of the `catch :exit` block. More on that [here](http://phrogz.net/programmingruby/tut_exceptions.html) (bottom of the page).

Reply:
Got it